src-built-in/components/advancedAppLauncher/src/components/FoldersList.jsx

```diff
@@ -212,21 +243,41 @@ export default class FoldersList extends React.Component {
 			className += ' folder-with-icon'
 		}
 
-		const EDITABLE_FOLDER_ICON_CLASS = 'ff-adp-hamburger'
+		const canDelete = folder.canDelete;
+		const canEdit = folder.canEdit;
+
+		let isEditing = false;
 
 		let nameField;
-		if (folder.icon === EDITABLE_FOLDER_ICON_CLASS && this.state.renamingFolder === folderName) {
+		if (this.state.renamingFolder === folderName && canEdit) {
 			nameField = <input id="rename" value={this.state.folderNameInput}
 			onChange={this.changeFolderName}
-			onKeyPress={this.keyPressed} className={this.state.isNameError ? "error" : ""} autoFocus />;
+			onKeyPress={this.keyPressed} className={this.state.isNameError ? "error" : ""} autoFocus ref={this.state.isNameError ? this.errorInput : null} />;
+			isEditing = true;
 		} else if (folderName === "Advanced App Launcher") {
 			nameField = "App Launcher"
 		} else {
 			nameField = folderName;
 		}
 
-		const canDelete = folder.canDelete;
-		const canEdit = folder.canEdit;
+		let buttons = null;
+		if (canEdit || canDelete) {
+			if (isEditing) {
+				buttons = (
+					<span className='folder-action-icons'>
+						{canDelete && <i id='confirm-edit' className='ff-check-mark-2' title='Accept Rename' onClick={this.renameFolder.bind(this, folderName)}></i>}
```

Comment:
Shouldn't this line check for canEdit instead of canDelete?

Reply:
Yup, switched the buttons at the last second cause the mockup had them in reverse locations. Forgot to change the boolean checks

Reply:
Since `isEditing` cannot be true without `canEdit` being true I don't even think that check is necessary at all on that line